### ISSUE TITLE
Weird Function Parameters Bug (2.2-dev30+)

### ISSUE THREAD

**Issue by Blueyescat** [bensku/Skript#842]
**(Read comment lines)**
```perl
function test(p1: text, p2: text, p3: text, p4: text, p5: text, p6: text, p7: text, p8: text, p9: text):
	broadcast "ok"

on load:
	#Works
	test("a", "a", "a", "a", "a", "a", "a", "a", "a")

	# ###########################
	#No error while parsing, but doesn't run function
	test("a", "a", "a", "a", "a", "a", "a", "a",  join {_x::*} with "")
	# ###########################

	# ###########################
	#These work fine
	test("a", "a", "a", "a", "a", "a", "a", "a",  "%join {_x::*} with """"%")
	test("a", "a", "a", "a", "a", "a", "a",  "%join {_x::*} with """"%", "a")
	# ###########################Ü

	test("a", "a", "a", "a", "a", "a", "a", join {_x::*} with "", "a")
	#The function 'test' requires at least 9 arguments, but only 8 are given.
	
	test("a", "a", "a", "a", "a", "a", join {_x::*} with "", "a", "a")
	#7 are given.
	
	test("a", "a", "a", "a", join {_x::*} with "", "a", "a", "a", "a")
	#5 are given.
	
	test(join {_x::*} with "", "a", "a", "a", "a", "a", "a", "a", "a")
	#1 is given.
	
	test("a", "a", "a", "a", "a", "a", "a", join {_x::*}, "a")
	#8 are given.
	
	test(join {_x::*}, "a", "a", "a", "a", "a", "a", "a", "a")
	#1 is given.
```
So, Skript can't read parameters **after** join expression. There is no problem with different parameter count/types and split expression.

**Comment by bensku** [comment]
Join is not supposed to produce lists, very interesting...

**Comment by Blueyescat** [comment]
Note: Tested with dev31c, the same.

**Comment by Syst3ms** [comment]
@bensku It's just that list parsing is being a little too lenient, taking function arguments as if they were part of a list

**Comment by TheLimeGlass** [comment]
https://github.com/bensku/Skript/issues/1148

**Comment by Blueyescat** [comment]
You don't need to reference a duplicate, also aready referenced at there.

**Comment by TheLimeGlass** [comment]
Well no one is going to close one so...

**Comment by bensku** [comment]
Not completely fixed, after all.

1. Still works as before.
2. Refer to #1089. In your case, last argument is null. Might fix this for next release, but separate issue.
3. Still works.
4. Still works.
5. Refer to #1089.
6. Refer to #1089.
7. Refer to #1089.
8. Refer to #1089.
9. Parsing fails, since
```
join {_x::*}, "a"
```
is valid expression. Maybe try parentheses?
10. Same as 9.

**Comment by bensku** [comment]
Went ahead and "solved" #1089 for next release.
